akka-http/src/main/scala/akka/http/javadsl/server/RejectionHandler.scala

```diff
@@ -1,190 +1,57 @@
-   * an `IllegalArgumentException`. Usually that means that the request content was not of the expected format.
-   * Note that semantic issues with the request content (e.g. because some parameter was out of range)
-   * will usually trigger a `ValidationRejection` instead.
-   */
-  def handleMalformedRequestContentRejection(ctx: RequestContext, message: String, cause: Throwable): RouteResult = passRejection()
-
-  /**
-   * Callback called to handle rejection created by unmarshallers.
-   * Signals that the request was rejected because an message body entity was expected but not supplied.
-   */
-  def handleRequestEntityExpectedRejection(ctx: RequestContext): RouteResult = passRejection()
-
-  /**
-   * Callback called to handle rejection created by marshallers.
-   * Signals that the request was rejected because the service is not capable of producing a response entity whose
-   * content type is accepted by the client.
-   */
-  def handleUnacceptedResponseContentTypeRejection(ctx: RequestContext, supported: jl.Iterable[String]): RouteResult = passRejection()
-
-  /**
-   * Callback called to handle rejection created by encoding filters.
-   * Signals that the request was rejected because the service is not capable of producing a response entity whose
-   * content encoding is accepted by the client
-   */
-  def handleUnacceptedResponseEncodingRejection(ctx: RequestContext, supported: jl.Iterable[HttpEncoding]): RouteResult = passRejection()
-
-  /**
-   * Callback called to handle rejection created by an [[akka.http.scaladsl.server.authentication.HttpAuthenticator]].
-   * Signals that the request was rejected because the user could not be authenticated. The reason for the rejection is
-   * specified in the cause.
-   *
-   * If credentialsMissing is false, existing credentials were rejected.
-   */
-  def handleAuthenticationFailedRejection(ctx: RequestContext, credentialsMissing: Boolean, challenge: HttpChallenge): RouteResult = passRejection()
-
-  /**
-   * Callback called to handle rejection created by the 'authorize' directive.
-   * Signals that the request was rejected because the user is not authorized.
-   */
-  def handleAuthorizationFailedRejection(ctx: RequestContext): RouteResult = passRejection()
-
-  /**
-   * Callback called to handle rejection created by the `cookie` directive.
-   * Signals that the request was rejected because a cookie was not found.
-   */
-  def handleMissingCookieRejection(ctx: RequestContext, cookieName: String): RouteResult = passRejection()
-
-  /**
-   * Callback called to handle rejection created when a websocket request was expected but none was found.
-   */
-  def handleExpectedWebSocketRequestRejection(ctx: RequestContext): RouteResult = passRejection()
+  def newBuilder = new RejectionHandlerBuilder(server.RejectionHandler.newBuilder)
+  
+  def defaultHandler = RejectionHandler(server.RejectionHandler.default)
+}
 
+case class RejectionHandler(unwrap: server.RejectionHandler) {
```

Comment:
does not need to be a case class, also, likely can be `final`. (anything case class is harder to evolve in bin compat way, so let's try to not use them where it's not a datatype)